applications/nrf_desktop/doc/motion.rst

```diff
@@ -3,6 +3,122 @@
+
+Simulated
+=========
+
+Selecting ``CONFIG_DESKTOP_MOTION_SIMULATED_ENABLE`` includes ``src/hw_interface/motion_simulated.c``.
+
+The ``motion`` module registers a shell module ``motion_sim`` and links to it
+two commands: ``start`` and ``stop``. When started the module will generate
+simulated motion events on the predefined trajectory (tracking an eight-sided
+polygon). The option ``CONFIG_DESKTOP_MOTION_SIMULATED_EDGE_TIME`` sets the
+edge crossing time while the option ``CONFIG_DESKTOP_MOTION_SIMULATED_SCALE_FACTOR``
+scales the size of the polygon.
+
+Implementation details
+**********************
+
+Implementation details describe the ``motion`` module implementation for motion
+sensors.
+
+Motion sensor
+=============
+
+The ``motion`` module sample data from the motion sensor using the motion
+sensor driver. Depending on the selected sensor name of a device linked
+with the driver changes. The same is true for sensor configuration options.
+The ``nrf_desktop`` application aggregates and translates sensor specific
+information to the application abstracts in ``configuration/common/motion_sensor.h``
+header file.
+
+Sampling thread
+===============
+
+The ``motion`` module uses a dedicated thread to sample data from the motion
+sensor. The reason for using the sampling thread is long time required for
+data to be ready. Using a thread also greatly simplifies the module as interaction
+with the sensor becomes sequential. It is assumed that data sampling will be
+done in the background and so the sampling thread priority is set to 0
+(the highest preemptive thread priority).
+
+The sampling thread stays in unready state blocked on a semaphore. The semaphore
+is triggered when the motion sensor trigger notifies data is available or
+other application event requires the module interaction with the sensor
+(e.g. when configuration is submitted from the host).
+
+Sampling pipeline
+=================
+
+The ``motion`` module starts in ``STATE_DISABLED`` and after initialization
+enters ``STATE_DISCONNECTED``. When disconnected module is not generating
+motion events but motion sensor is sampled to make sure its registers are cleared.
+
+Upon connection ``motion`` module switches to ``STATE_IDLE`` in this state
+module waits for the motion sensor trigger. When first motion is detected and
+the device is connected to host (can transmit the HID data out) module switches
+to ``STATE_FETCHING``, samples the motion data and submits the ``motion_event``.
+The ``motion`` module then waits for indication that the ``motion_event`` data
+was transmitted to the host. This is done when the module receives the
+``hid_report_sent_event`` event. At that moment a next motion sampling is
+performed and the next ``motion_event`` sent.
+
+The module continues to sample data until disconnection or when there is no
+motion detected. The ``motion`` module assumes no motion when ``10`` consecutive
+samples return zero on both axis.
```

Comment:
What happns then? It goes back to the IDLE state?

Reply:
added